File: Web/Scout.Web.UI/src/model/player.model.ts
class PlayerStatistics {
    public playerIdentifier: string;
    public teamIdentifier: string;
    public teamName: string;
    public teamYear: number;
}

export class PlayerBattingStatistics extends PlayerStatistics {
    public plateAppearances: number;
    public atBats: number;
    public hits: number;
    public doubles: number;
    public triples: number;
    public homeruns: number;
    public runsBattedIn: number;
    public sacrificeHits: number;
    public sacrificeFlies: number;
    public walks: number;
    public intentionalWalks: number;
    public hitByPitch: number;
    public strikeouts: number;
    public stolenBases: number;
    public caughtStealing: number;
    public groundedIntoDoublePlay: number;
    public battingAverage: number;
    public onBasePercentage: number;
    public sluggingPercentage: number;
    public onBasePlusSlugging: number;
    public onBasePlusSluggingAdj: number;
}

export class PlayerAdvancedBattingStatistics extends PlayerStatistics {
    public battingAverage: number;
    public totalAverage: number;
    public onBasePercentage: number;
    public sluggingPercentage: number;
    public onBasePlusSlugging: number;
    public onBasePlusSluggingAdj: number;
    public battingAvgOfBallsInPlay: number;
    public homeRunPercentage: number;
    public strikeoutPercentage: number;
    public walkPercentage: number;
    public extraBaseHitPercentage: number;
    public inPlayPercentage: number;
}

export class PlayerPitchingStatistics extends PlayerStatistics {
    public pitchingStint: number;
    public gamesWon: number;
    public gamesLost: number;
    public gamesPlayed: number;
    public completeGames: number;
    public shutouts: number;
    public gamesSaved: number;
    public inningsPitched: number;
    public hits: number;
    public runs: number;
    public earnedRuns: number;
    public walks: number;
    public strikeouts: number;
    public homeruns: number;
    public earnedRunAverage: number;
    public intentionalWalks: number;
    public hitBatsmen: number;
    public wildPitches: number;
    public balks: number;
    public timesInducedGidp: number;
}

export class Player {
    public playerId: number;
    public playerIdentifier: string;
    public retrosheetId: string;
    public firstName: string;
    public lastName: string;
    public birthDate?: Date;
    public birthCity?: string;
    public birthStateProvince?: string;
    public birthCountry?: string;
    public deathDate?: Date;
    public deathCity: string;
    public deathStateProvince: string;
    public deathCountry: string;
    public primaryPosition: string;
    public draftTeamId?: number;
    public draftYear?: number;
    public majorLeagueDebut?: number;
    public bats: string;
    public throws: string;
    public height?: number;
    public weight?: number;

    public battingStatistics: PlayerBattingStatistics[];
    public pitchingStatistics: PlayerPitchingStatistics[];
    public advancedBattingStatistics: PlayerAdvancedBattingStatistics[];
}

export class PlayerListItem {
    public playerCode: string;
    public firstName: string;
    public lastName: string;
    public playerRetrosheetId: string;
}

File: Model/Scout.Model.DB/PlayerModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Scout.Model.DB
{
    public class PlayerModel : ScoutModel
    {
        [ConcurrencyCheck]
        [Column(TypeName = "varchar(16)"), Required]
        public string PlayerIdentifier { get; set; }
        [Column(TypeName = "varchar(16)"), Required]
        public string RetrosheetId { get; set; }
        [Column(TypeName = "varchar(32)"), Required]
        public string FirstName { get; set; }
        [ConcurrencyCheck]
        [Column(TypeName = "varchar(32)"), Required]
        public string LastName { get; set; }
        public Nullable<System.DateTime> Birthdate { get; set; }
        [Column(TypeName = "varchar(64)"), Required]
        public string BirthCity { get; set; }
        [Column(TypeName = "varchar(64)")]
        public string BirthStateProvince { get; set; }
        [Column(TypeName = "varchar(64)"), Required]
        public string BirthCountry { get; set; }
        public Nullable<System.DateTime> DeathDate { get; set; }
        [Column(TypeName = "varchar(64)")]
        public string DeathCity { get; set; }
        [Column(TypeName = "varchar(64)")]
        public string DeathStateProvince { get; set; }
        [Column(TypeName = "varchar(64)")]
        public string DeathCountry { get; set; }
        public Nullable<System.DateTime> MajorLeagueDebut { get; set; }
        [Column(TypeName = "varchar(1)")]
        public string Bats { get; set; }
        [Column(TypeName = "varchar(1)")]
        public string Throws { get; set; }
        public Nullable<short> Height { get; set; }
        public Nullable<short> Weight { get; set; }

        public virtual ICollection<PlayerPitchingStatisticsModel> PlayerPitchingStatistics { get; set; }
        public virtual ICollection<PlayerBattingStatisticsModel> PlayerBattingStatistics { get; set; }
        public virtual ICollection<PlayerFieldingStatisticsModel> PlayerFieldingStatistics { get; set; }
    }
}
